ch_tools/common/logging.py

```diff
@@ -0,0 +1,213 @@
+        """
+
+        return record["extra"].get("logger_name") == self._name
+
+
+def make_filter(name):
+    """
+    Factory for filter creation.
+    """
+
+    return Filter(name)
+
+
+class InterceptHandler(logging.Handler):
+    """
+    Helper class for logging interception.
+    """
+
+    def emit(self, record: logging.LogRecord) -> None:
+        """
+        Intercept all records from the logging module and redirect them into loguru.
+
+        The handler for loguru will be chosen based on module name.
+        """
+
+        # Get corresponding Loguru level if it exists.
+        level: int or str  # type: ignore
+        try:
+            level = logger.level(record.levelname).name
+        except ValueError:
+            level = record.levelno
+
+        # Find caller from where originated the logged message.
+        frame, depth = inspect.currentframe(), 0
+        while frame and (depth == 0 or frame.f_code.co_filename == logging.__file__):
+            frame = frame.f_back
+            depth += 1
+
+        logger.bind(logger_name=record.name).opt(
+            depth=depth, exception=record.exc_info
+        ).log(level, record.getMessage())
+
+
+def configure(
+    config_loguru: dict,
+    module: str,
+    extra: Optional[dict] = None,
+    opt: Optional[dict] = None,
+) -> None:
+    """
+    Configure logger.
+    """
+    logger_config["module"] = module
+    logger_config["extra"] = extra if extra else {}
+    logger_config["opt"] = opt if opt else {}
+    loguru_handlers = []
+
+    for name, value in config_loguru["handlers"].get(module, {}).items():
+        format_ = config_loguru["formatters"][value["format"]]
+        handler = {
+            "sink": value["sink"],
+            "format": format_,
+            "enqueue": True,
+            "diagnose": False,
+        }
+        if "level" in value:
+            handler["level"] = value["level"]
+
+        if "filter" in value:
+            if value["filter"]:
+                handler["filter"] = make_filter(value["filter"])
+                # https://loguru.readthedocs.io/en/stable/api/logger.html#message
+                # One can also pass a dict mapping module names to minimum required level. In such case, each log record will search for it’s closest parent in the dict
+                # and use the associated level as the filter.
+                # In order to set a default level, the "" module name should be used as it is the parent of all modules (it does not suppress global level threshold, though).
+                handler["filter"][""] = False
+        else:
+            handler["filter"] = make_filter(name)
+        loguru_handlers.append(handler)
+
+    logger.configure(handlers=loguru_handlers, activation=[("", True)])
+
+    # Configure logging.
+    logging.basicConfig(handlers=[InterceptHandler()], level=0)
+
+
+def _log(level, msg, *args, **kwargs):
+    # Make a copy so we can rewrite exception option
+    opt = dict(logger_config["opt"])
```

Comment:
afaik the opt is used only for single logging call. So it make sense to remove it from the global state

Reply:
What did you mean by removing it from the global state? `logger_config["module"]` is also used just for logging call, passing these values from configure to a logging call is a reason why we store them in a dict.

Reply:
I mean  remove  `opt` from  `logger_config`.  Because it never used and also everything that we can store in it, we should pass thought `logger_config['extra']`